### PR TITLE
Add past CTFs user role

### PR DESCRIPTION
**Pull request by JJ-8**
As discussed in https://github.com/TFNS/CTFNote/issues/68
Ported from:
- https://github.com/JJ-8/CTFNote/pull/6/commits/512425e03b71ce55efc50aab8eb74451f9719252
- https://github.com/JJ-8/CTFNote/pull/8/commits/855c1d49ee3517ef6460a50a3d1ccf38a20d1c60

### CODE REVIEW

**Comment by XeR** [comment]
> "Past CTFs" is the only accurate description of the role I could think of. Do you have any suggestions?

The problem with the "past CTF" name is that it does not fit new roles. It would have been a good fit in the previous version since we used rights (`CTF_ALL`, `EDIT_CTF`, etc.).
@B-i-t-K mentioned "friends" last week. He had an other idea but I'm afraid I can't remember it.
Perhaps he'll chime in ?

> it is pretty difficult to change the name in SQL.

We do not need to undo changes that did not reach the `main` branch. (it's fine in this case since it's not merged yet)
Once they reach main, we should consider the migrations to be set in stone.

If you mean "how do I fix my prod database now that I've applied the change", I do not know. Surely, you can dump the database, patch the SQL file, wipe the volume and import it if everything else fails.

**Comment by JJ-8** [comment]
@XeR @B-i-t-K 

Is the name of the role the only thing that blocks this PR from merging? If so, can we fix that at a later stage? Just creating a new PR for changing the name of this role to your liking?

It would really be nice if this last PR gets merged, because then I can start upgrading my CTFNote instance too :)

**Comment by B-i-t-K** [comment]
Sorry for the delay.

Yeah just change the name from "Past ctf" to "Friend" and it's good to merge!

**Comment by JJ-8** [comment]
@XeR @B-i-t-K Done!

File: api/migrations/38-add-friend-user.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+--Check if current user is allowed to access past ctfs
+CREATE FUNCTION ctfnote_private.is_friend ()
+  RETURNS boolean
+  AS $$
+  SELECT
+    ctfnote_private.current_role () = 'user_friend'::ctfnote.role
+    OR ctfnote_private.is_member ();
```
Comment (XeR) [inline api/migrations/38-add-friend-user.sql:7]: Shouldn't we redefine `is_guest` to check for `is_friend` instead of `is_member` ?

Comment (JJ-8) [reply api/migrations/38-add-friend-user.sql:7]: You are totally right. Good catch!

**Review by XeR — APPROVED** [review]
LGTM

**Pull request merged**